cf-check/ts_key_read.c

```diff
@@ -0,0 +1,85 @@
+#include <platform.h>
+#include <ts_key_read.h>
+#include <logging.h>
+#include <definitions.h> // CF_BUFSIZE
+#include <known_dirs.h> // GetStateDir()
+#include <file_lib.h> // FILE_SEPARATOR
+#include <alloc.h> // xstrdup()
+
+void ReadMonitoringSlotsMeta(const char *ts_key_path, Seq *slotsMeta)
+{
+    char buf[CF_BUFSIZE];
+    const char *filename;
+    int i;
+
+    if (ts_key_path == NULL)
+    {
+        snprintf(buf, CF_BUFSIZE - 1, "%s%cts_key", GetStateDir(), FILE_SEPARATOR);
+        filename = buf;
+    }
+    else
+    {
+        filename = ts_key_path;
+    }
+
+    FILE *f = safe_fopen(filename, "r");
+    if (f == NULL)
+    {
+        perror("open"); // TODO improve error message
```

Comment:
TODO, this wasn't handled well before but should be now. Maybe fix libpromises/monitoring_read.c as well.